packages/gatsby-plugin-mdx/gatsby/source-nodes.js

```diff
@@ -131,19 +146,17 @@ module.exports = (
         },
         async resolve(mdxNode, { pruneLength }) {
           if (mdxNode.excerpt) {
-            return Promise.resolve(mdxNode.excerpt)
+            const mdxExcerptNode = await createMDXNode({
```

Comment:
`gray-matter` can't interpret any `import` or `export` statement. So, if we use the string excerpt, it could contain unnecessary text.

I'm ok with it for my use case, though. I'll revert this.